### DISCUSSION TITLE
Misaligned entities ('-') will be ignored during training - warning

### DISCUSSION THREAD

**Discussion by NaziaToma** [explosion/spaCy#9161] [Help: Coding & Implementations] [answered]
I am getting this warning -

UserWarning: [W030] Some entities could not be aligned in the text "Dhaka Mina Para Eidgah House" with entities "[[0, 21, 'BUSINESS_NAME'], [23, 28, 'HOLDING_NUMBE...". Use `spacy.training.offsets_to_biluo_tags(nlp.make_doc(text), entities)` to check the alignment. Misaligned entities ('-') will be ignored during training.

Is it going to affect my accuracy of the model? How do I resolve it? It's showing for all of the dataset.

**Comment by polm** [comment, marked as answer]
This happens when your entity annotations don't line up with token boundaries. In the example you gave:

```
text = "Dhaka Mina Para Eidgah House"
text[0:21] -> 'Dhaka Mina Para Eidga' # no "h"
text[23:28] -> 'House' # this looks fine
```

In the first case, because the right edge of the entity is halfway through a token it's not clear what the right thing to do is (you can't put an entity label on half a token). 

If this is happening for all of your annotations then something is wrong with the way you generated, maybe you have an off-by-one error in your math somewhere? You'd need to look at how they're generated.
